public/app/plugins/datasource/azuremonitor/dataquery.gen.ts

```diff
@@ -0,0 +1,303 @@
+}
+
+export interface AzureMetricDimension {
+  /**
+   * Name of Dimension to be filtered on.
+   */
+  dimension?: string;
+  /**
+   * @deprecated filter is deprecated in favour of filters to support multiselect.
+   */
+  filter?: string;
+  /**
+   * Values to match with the filter.
+   */
+  filters?: Array<string>;
+  /**
+   * String denoting the filter operation. Supports 'eq' - equals,'ne' - not equals, 'sw' - starts with. Note that some dimensions may not support all operators.
+   */
+  operator?: string;
+}
+
+export const defaultAzureMetricDimension: Partial<AzureMetricDimension> = {
+  filters: [],
+};
+
+export type GrafanaTemplateVariableQueryType = ('AppInsightsMetricNameQuery' | 'AppInsightsGroupByQuery' | 'SubscriptionsQuery' | 'ResourceGroupsQuery' | 'ResourceNamesQuery' | 'MetricNamespaceQuery' | 'MetricNamesQuery' | 'WorkspacesQuery' | 'UnknownQuery');
+
+export interface BaseGrafanaTemplateVariableQuery {
+  rawQuery?: string;
+}
+
+export interface UnknownQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'UnknownQuery';
+}
+
+export interface AppInsightsMetricNameQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'AppInsightsMetricNameQuery';
+}
+
+export interface AppInsightsGroupByQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'AppInsightsGroupByQuery';
+  metricName: string;
+}
+
+export interface SubscriptionsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'SubscriptionsQuery';
+}
+
+export interface ResourceGroupsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'ResourceGroupsQuery';
+  subscription: string;
+}
+
+export interface ResourceNamesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'ResourceNamesQuery';
+  metricNamespace: string;
+  resourceGroup: string;
+  subscription: string;
+}
+
+export interface MetricNamespaceQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricNamespaceQuery';
+  metricNamespace?: string;
+  resourceGroup: string;
+  resourceName?: string;
+  subscription: string;
+}
+
+/**
+ * @deprecated Use MetricNamespaceQuery instead
+ */
+export interface MetricDefinitionsQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricDefinitionsQuery';
+  metricNamespace?: string;
+  resourceGroup: string;
+  resourceName?: string;
+  subscription: string;
+}
+
+export interface MetricNamesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'MetricNamesQuery';
+  metricNamespace: string;
+  resourceGroup: string;
+  resourceName: string;
+  subscription: string;
+}
+
+export interface WorkspacesQuery extends BaseGrafanaTemplateVariableQuery {
+  kind: 'WorkspacesQuery';
+  subscription: string;
+}
+
+export type GrafanaTemplateVariableQuery = (AppInsightsMetricNameQuery | AppInsightsGroupByQuery | SubscriptionsQuery | ResourceGroupsQuery | ResourceNamesQuery | MetricNamespaceQuery | MetricDefinitionsQuery | MetricNamesQuery | WorkspacesQuery | UnknownQuery);
+
+export interface AzureMonitor {}
```

Comment:
So the actual data source query generated from schema is not used?

Reply:
I can use the type, however for the sake of readable code the name `AzureMonitor` doesn't make sense. This name should be `AzureMonitorQuery` but I can't tell how to rename the default.

Reply:
@sdboyer shouldn't data query names indeed end in "Query"?